src/util/finite_field_value.h

```diff
@@ -31,26 +31,45 @@
 
 namespace cvc5::internal {
 
+/** Stores a field size that have been validated to be prime */
+struct FfSize
+{
+  FfSize(Integer size) : d_val(size)
+  {
+    // we only support prime fields right now
+    Assert(size.isProbablePrime()) << "not prime: " << size;
+  }
+  FfSize(int size) : d_val(size)
+  {
+    // we only support prime fields right now
+    Assert(d_val.isProbablePrime()) << "not prime: " << size;
+  }
+  FfSize(size_t size) : d_val(size)
+  {
+    // we only support prime fields right now
+    Assert(d_val.isProbablePrime()) << "not prime: " << size;
+  }
+  operator const Integer&() const { return d_val; }
+  bool operator==(const FfSize& y) const { return d_val == y.d_val; }
+  bool operator!=(const FfSize& y) const { return d_val != y.d_val; }
+
+  Integer d_val;
+}; /* struct FfSize */
+
 class FiniteFieldValue
 {
  public:
-  FiniteFieldValue(const Integer& val, const Integer& size)
+  FiniteFieldValue(const Integer& val, const FfSize& size)
```

Comment:
This is the key change re: `FfSize`. Without this, we end up doing a tone of extra primality tests.